include/circt/Dialect/StaticLogic/StaticLogic.td

```diff
@@ -0,0 +1,126 @@
+//===- StaticLogic.td - StaticLogic Definitions ------------*- tablegen -*-===//
+//
+// Copyright 2020 The CIRCT Authors.
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//===----------------------------------------------------------------------===//
+
+#ifdef STATICLOGIC_OPS
+#else
+#define STATICLOGIC_OPS
+
+#ifdef OP_BASE
+#else
+include "mlir/IR/OpBase.td"
+#endif // OP_BASE
+
+include "mlir/IR/SymbolInterfaces.td"
+include "mlir/IR/RegionKindInterface.td"
+include "mlir/Interfaces/CallInterfaces.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+def StaticLogic_Dialect : Dialect {
+  let name = "staticlogic";
+  let cppNamespace = "";
+}
+
+def PipelineOp : Op<StaticLogic_Dialect, "pipeline", [FunctionLike, Symbol,
+  NativeOpTrait<"IsIsolatedFromAbove">
+]> {
+  let summary = "pipeline operation";
+  let description = [{
+    The "staticlogic.pipeline" operation represents a statically scheduled 
+    pipeline stucture which can contain multiple pipeline stages.
```

Comment:
I think the key thing here is that pipeline stages are represented by MLIR Blocks.

Reply:
fixed.